### ISSUE TITLE
Options are hard to set

### ISSUE THREAD

**Issue by nonhermitian** [Qiskit/qiskit-ibm-runtime#543]
**What is the expected feature or enhancement?**

The `Options` object is a bit confusing, and the setting of certain parameters is more work than it should be.  The `Options` class currently has the form:

```python

Options(optimization_level=1, resilience_level=0, log_level='WARNING', transpilation=Transpilation(skip_transpilation=False, initial_layout=None, layout_method=None, routing_method=None, translation_method=None, approximation_degree=None, timing_constraints=None, seed_transpiler=None), execution=Execution(shots=4000, qubit_lo_freq=None, meas_lo_freq=None, rep_delay=None, init_qubits=True), experimental=None)
```
Indicating that there are primitive wide options such as `optimization_level`, `resilience_level` etc, and specific sets of options related to transpilation and execution.  It is a bit confusing because `optimization_level` is a transpiler keyword argument but is not in the transpilation options.  

However, more importantly it is difficult to set the options for execution and transpilation in a straightforward way.  Either one has to import the `Transpilation` and `Execution` classes, which are not top level imports and forces users to learn yet another class, or one must modify arguments via accessing attributes. E.g.

```python

options = Options()
options.transpilation.skip_transpilation=True
```

What this means is that if I am not directly setting these options, eg wrapping some functionality into a call with a primitive, I need to loop over values of dict or place all the kwargs into a signature and then set all of those values for each attribute the object has.  Alternatively it would be nice to have something like 

```python
options.transpilation.set_options(trans_options_dict)
```
so that users can just use a plain dict to set options. Or just allow plain dicts in the call to `Options` in the first place rather than requiring the internal class types.

**Comment by jyu00** [comment]
@nonhermitian What do you think about passing dictionaries into `Options`? Something like

```
options = Options(transpilation={"skip_transpilation": True"})
```

I had also entertain the thought of accepting `kwargs`. For example, 
```
Options(skip_transpilation=True)
```
would be the same as setting `transpilation.skip_transpilation=True`.

**Comment by jyu00** [comment]
`optimization_level` is at level 1 because it's considered a frequently used keyword. Moving it to be under transpilation makes it more difficult to set. Having it in both places seems more confusing.

**Comment by nonhermitian** [comment]
Yeah, that would work.

**Comment by jyu00** [comment]
Fixed by #533. You can now pass a dictionary to each option group, e.g. 

```
options = Options(transpilation={"skip_transpilation": True"})
```
